pkg/build/builder/util.go

```diff
@@ -55,11 +58,65 @@ func getDockerNetworkMode() stiapi.DockerNetworkMode {
 	defer file.Close()
 
 	if id := readNetClsCGroup(file); id != "" {
-		return stiapi.NewDockerNetworkModeContainer(id)
+		return s2iapi.NewDockerNetworkModeContainer(id)
 	}
 	return ""
 }
 
+// GetCGroupLimits returns a struct populated with cgroup limit values gathered
+// from the local /sys/fs/cgroup filesystem.  Overflow values are set to
+// math.MaxInt64.
+func GetCGroupLimits() (*s2iapi.CGroupLimits, error) {
+	byteLimit, err := readInt64("/sys/fs/cgroup/memory/memory.limit_in_bytes")
+	if err != nil {
+		return nil, err
+	}
+
+	cpuQuota, err := readInt64("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us")
+	if err != nil {
+		return nil, err
+	}
+
+	cpuShares, err := readInt64("/sys/fs/cgroup/cpu,cpuacct/cpu.shares")
+	if err != nil {
+		return nil, err
+	}
+
+	cpuPeriod, err := readInt64("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_period_us")
+	if err != nil {
+		return nil, err
+	}
+	return &s2iapi.CGroupLimits{
+		CPUShares:        cpuShares,
+		CPUPeriod:        cpuPeriod,
+		CPUQuota:         cpuQuota,
+		MemoryLimitBytes: byteLimit,
+		// Set memoryswap==memorylimit, this ensures no swapping occurs.
+		// see: https://docs.docker.com/engine/reference/run/#runtime-constraints-on-cpu-and-memory
+		MemorySwap: byteLimit,
+	}, nil
+}
+
+func readInt64(filePath string) (int64, error) {
+	data, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return -1, err
+	}
+	s := strings.TrimSpace(string(data))
+	val, err := strconv.ParseInt(s, 10, 64)
+	// overflow errors are ok, we'll get return a math.MaxInt64 value which is more
+	// than enough anyway.  For underflow we'll return MinInt64 and the error.
+	if err != nil && err.(*strconv.NumError).Err == strconv.ErrRange {
+		if s[0] == '-' {
+			return math.MinInt64, err
+		}
+		return math.MaxInt64, nil
```

Comment:
@rhcarvalho unless you can provide a scenario in which the above code does the wrong thing in terms of what limit ultimately gets enforced, i'd like to stop debating this.  I don't see a benefit from what you propose, and "0" is a pretty ambiguous value to return, you're relying on downstream apis (cgroup and docker) to interpret it in a particular way.  maxint64 has no such requirements because it actually asserts a specific limit which effectively represents no limit.

furthermore the value returned from underflow is irrelevant since we're returning an error, so the returned value is basically undefined.